dig.go

```diff
@@ -235,13 +250,13 @@ func (c *Container) get(t reflect.Type) (reflect.Value, error) {
 	// Provide-time validation ensures that all constructors return at least
 	// one value.
 	if err := constructed[len(constructed)-1]; err.Type() == _errType && err.Interface() != nil {
-		return _noValue, fmt.Errorf("constructor %v for type %v failed: %v", n.ctype, t, err.Interface())
+		return _noValue, fmt.Errorf("constructor %v for type %v failed: %v", n.ctype, e.t, err.Interface())
 	}
 
 	for _, con := range constructed {
-		c.set(con)
+		c.set(key{t: con.Type()}, con)
```

Comment:
This doesn't seem right. If I did `get(key{name: "foo", t: someType})`, this will `set(key{t: someType})`.